services/middlewareServices/FormSGService.ts

```diff
@@ -0,0 +1,57 @@
+import FormSG from "@opengovsg/formsg-sdk"
+import { Request, Response, NextFunction } from "express"
+
+import logger from "@logger/logger"
+
+import { AuthError } from "@root/errors/AuthError"
+
+const formsg = FormSG()
+
+export default class FormSGService {
+  authenticate = () => (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): void => {
+    const signature = req.get("X-FormSG-Signature")
+    if (!signature) {
+      throw new AuthError("Signature missing")
+    }
+    const postUri = `https://${req.get("host")}${req.baseUrl}${req.path}`
+    try {
+      formsg.webhooks.authenticate(signature, postUri)
+      // Continue processing the POST body
+      next()
+    } catch (e) {
+      logger?.error(e)
+      throw new AuthError("Unauthorized")
+    }
+  }
+
+  decrypt = ({ formKey }: { formKey: string }) => (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): void => {
+    try {
+      const submission = formsg.crypto.decrypt(
+        formKey,
+        // If `verifiedContent` is provided in `req.body.data`, the return object
+        // will include a verified key.
+        req.body.data
+      )
+
+      // If the decryption failed, submission will be `null`.
+      if (submission) {
+        // Continue processing the submission
+        res.locals.submission = submission
+        next()
+      } else {
+        res.status(422).send({ message: "Bad submission" })
+      }
```

Comment:
nit: we can probably invert the order and save a level of indent

Reply:
Makes sense, https://github.com/isomerpages/isomercms-backend/pull/458/commits/292486b5631d525c3d3d9189758957c50c8c5dad

Reply:
is the `!submission` a strong enough check? if submission is an empty object, this might still pass (unless we're ok with an empty submission?) - unsure of details but just wanted to double confirm

Reply:
This should be fine - the documentation states that the submission will be `null` if the decryption failed? But to be safe I can explicitly check for null - 493c454

Reply:
actually checking for `null` is weaker because now it doesn't cover `undefined` HAHA on opening it in vscode, i see that hte type is `DecryptedContent | null`, which meant that your prev check/this check both works.